### ISSUE TITLE
How can I make it work with linux?

### ISSUE THREAD

**Issue by GrasSoft** [BlackStartx/PyCharm-Blender-Plugin#29]
I tried setting up the executable as per the tutorial provided. The search does not see linux binaries. Even after I add the path to the binary manually it does not start blender from PyCharm.

Thank you!

**Comment by Doomerdinger** [comment]
I made some changes on a fork in the past that allow for Linux (and Mac) to run.
Bit out of date with the head branch.
https://github.com/Doomerdinger/PyCharm-Blender-Plugin/tree/ubuntu-updates

Works for me on Ubuntu 22.04.

I'll update it and dig up the plugin file for you at some point if you're unable to generate it yourself.

**Comment by BlackStartx** [comment]
Hi @Doomerdinger~ 🌺

First of all thanks so much for the fork on my behalf!
I never setted up an environment to deploy and test Mac & Linux builds, so your branches are indeed very helpful!

From what I can see based on the commits, the main difference in the source code is the usage of '**`/`**' instead of '**`\`**' for the paths.

I also see changes to the Gradle files, but I guess those are only related to compiling?
> Do correct me if I'm wrong~ ^-^

## About the branches

If you want I can integrate your changes into the main branch, if that's the case feel free to make a pull request!

We can tweak it to make the built plugin compatible with multiple OS at the same time (Windows, Mac & Linux) and I'll make sure to ship it in the next update~ ✨

Thanks in advance,
- BlackStartx

**Comment by Doomerdinger** [comment]
It has been quite some time (years) since I figured all that out.
The slashes were definitely part of it.
I'll go back and try it again, see if I can dredge up my understanding of the pitfalls.

As a note, on Mac when I hit breakpoints in the debugger it opens up the symlinked file copy instead of the one in the source which is somewhat annoying since then the code completion is broken and the IDE gets confused about contents in mem vs on disk. Not sure what causes this, have not looked into it much.

**Comment by BlackStartx** [comment]
> As a note, on Mac when I hit breakpoints in the debugger it opens up the symlinked file copy instead of the one in the source

Yeah... I faced that issue a LOT in the first releases of the tool, kinda fixed with [Version 2023.2](https://github.com/BlackStartx/PyCharm-Blender-Plugin/commit/87c0b2b9f233783d9046c9bea023a565aef233db#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L34-L141), but it seems only for Windows OS.

Has also been reported on YouTrack:

- https://youtrack.jetbrains.com/issue/PY-34001
- https://youtrack.jetbrains.com/issue/PY-18491

A workaround was based on path separators (again '**/**' and '**\\**'), so I guess it make sense why it didn't worked on other OS~

**Comment by ilbozo** [comment]
i've found a workaround. Create a symbolic link of the blender executable as blender.exe and the file is shown in the ui.
